lib/ts/BufferWriterFormat.cc

```diff
@@ -211,24 +211,25 @@ namespace bw_fmt
           *dst++ = spec._fill;
         break;
       case BWFSpec::Align::CENTER:
-        d2 = (delta + 1) / 2;
-        if (d2 > 1) {
-          dst = base + d2; // move existing content to here.
-          if (dst < limit) {
-            last = dst + size; // amount of data to move.
-            if (last > limit)
-              last = limit;
-            std::memmove(dst, base, last - dst);
-          }
-          dst  = base + size + d2;
-          last = base + delta / 2;
+        d2 = (delta + 1) / 2; // always > 0 because min > extent
```

Comment:
why is `extent` used instead of `size`? couldn't the `delta` be overestimated since `extent` includes dropped characters. why are we accounting for dropped characters when we align?

Reply:
Two reasons.

* This is necessary to do two-pass string sizing (which is currently used for writing to a `std::string` and is expected to be use  later for logging). In such cases the overall extent can't be calculated if these local extents are not tracked correctly.
* The fact that the output went past the end of the buffer shouldn't affect the alignment. That should occur *as if* the buffer was infinite and then clipped to its actual length.